src/rpc/blockchain.cpp

```diff
@@ -583,8 +583,11 @@ static CBlock GetBlockChecked(BlockManager& blockman, const CBlockIndex& blockin
     CBlock block;
     {
         LOCK(cs_main);
-        if (blockman.IsBlockPruned(blockindex)) {
-            throw JSONRPCError(RPC_MISC_ERROR, "Block not available (pruned data)");
+        if (!(blockindex.nStatus & BLOCK_HAVE_DATA)) {
+            if (blockman.IsBlockPruned(blockindex)) {
+                throw JSONRPCError(RPC_MISC_ERROR, "Block not available (pruned data)");
+            }
+            throw JSONRPCError(RPC_MISC_ERROR, "Block not available (not fully downloaded)");
```

Comment:
nit:  The comment below should also be adjusted.  The case where the header is available but we don't have the block yet is covered by this new `JSONRPCError`.

        // Block not found on disk. This could be because we have the block
        // header in our index but not yet have the block or did not accept the
        // block. Or if the block was pruned right after we released the lock above.

Reply:
done, thanks!